### PR TITLE
Introduce `DripListManager`

### PR DESCRIPTION
**Pull request by jtourkos**
Resolves and context 👉 https://github.com/radicle-dev/drips-app-v2/issues/393

And the [meat🥩](https://github.com/radicle-dev/drips-app-v2/pull/447/files#diff-7604ec522e10f22306b0103490b932841b6a20fa644ab627e4aecb1197e53b47) of the changes.

### CODE REVIEW

File: src/lib/utils/driplist/DripListManager.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+export default class DripListManager {
+  private _nftDriverClient!: NFTDriverClient;
+  private _nftDriverTxFactory!: NFTDriverTxFactory;
+  private _addressDriverClient!: AddressDriverClient;
+  private readonly _dripsSubgraphClient = getSubgraphClient();
+  private readonly _nftDriverMetadataManager = new NftDriverMetadataManager();
+  private readonly _repoDriverMetadataManager = new RepoDriverMetadataManager();
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  private constructor() {}
```
Comment (efstajas) [inline src/lib/utils/driplist/DripListManager.ts:44]:
```suggestion

```

File: src/lib/utils/metadata/MetadataManagerBase.ts [hunk 2]
```diff
@@ -32,7 +34,7 @@ export interface IMetadataManager<TAccountMetadataSchema extends z.ZodType, TAcc
 
   updateAccountMetadata<T extends z.ZodType>(
     newData: z.infer<T>,
-    lastKnownHash: string | null,
+    lastKnownHash: string | undefined,
```
Comment (efstajas) [inline src/lib/utils/metadata/MetadataManagerBase.ts:37]:
```suggestion
    lastKnownHash: IpfsHash | undefined,
```

File: src/lib/utils/metadata/MetadataManagerBase.ts [hunk 3]
```diff
@@ -32,7 +34,7 @@ export interface IMetadataManager<TAccountMetadataSchema extends z.ZodType, TAcc
...
     schema: T,
   ): Promise<{ newHash: string; tx: ContractTransaction }>;
```
Comment (efstajas) [inline src/lib/utils/metadata/MetadataManagerBase.ts:39]:
```suggestion
  ): Promise<{ newHash: IpfsHash; tx: ContractTransaction }>;
```

File: src/lib/utils/metadata/MetadataManagerBase.ts [hunk 4]
```diff
@@ -160,7 +162,7 @@ export default abstract class MetadataManagerBase<
    */
   public async updateAccountMetadata<T extends z.ZodType>(
     newData: z.infer<T>,
-    lastKnownHash: string | null,
+    lastKnownHash: string | undefined,
```
Comment (efstajas) [inline src/lib/utils/metadata/MetadataManagerBase.ts:165]:
```suggestion
    lastKnownHash: IpfsHash | undefined,
```

File: src/lib/utils/metadata/MetadataManagerBase.ts [hunk 5]
```diff
@@ -160,7 +162,7 @@ export default abstract class MetadataManagerBase<
...
   ): Promise<{ newHash: string; tx: ContractTransaction }> {
```
Comment (efstajas) [inline src/lib/utils/metadata/MetadataManagerBase.ts:166]:
```suggestion
  ): Promise<{ newHash: IpfsHash; tx: ContractTransaction }> {
```

Comment (efstajas) [reply src/lib/utils/metadata/MetadataManagerBase.ts:166]: Probably some other strings that should use IpfsHash type in this file too

File: src/lib/utils/driplist/DripListManager.ts [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+import type { UserId } from '../metadata/types';
+import MetadataManagerBase from '../metadata/MetadataManagerBase';
+import type { DripsReceiverConfig } from 'radicle-drips';
+import type { BigNumberish, ContractTransaction } from 'ethers';
+import assert from '$lib/utils/assert';
+
+/**
+ * A class for managing `DripList`s.
+ *
+ * **Important**: This class assumes that *all* clients and factories are connected to the *same* singer.
```
Comment (efstajas) [inline src/lib/utils/driplist/DripListManager.ts:33]:
```suggestion
 * **Important**: This class assumes that *all* clients and factories are connected to the *same* signer.
```

🎤🎶

**Review by efstajas — APPROVED** [review]
Looks amazing! Some very minor comments :)

File: src/lib/utils/driplist/DripListManager.ts (hunk 1 above)
Comment (jtourkos) [reply src/lib/utils/driplist/DripListManager.ts:44]: Aren't those identical?👀

File: src/lib/utils/driplist/DripListManager.ts (hunk 6 above)
Comment (jtourkos) [reply src/lib/utils/driplist/DripListManager.ts:33]: 🤣

**Pull request merged**